Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3339,12 +3339,15 @@ extension BrowserViewController: TopTabsDelegate {
         openBlankNewTabAndFocus(isPrivate: isPrivate)
     }
     
-    func topTabsDidPressPrivateTab() {
+    func topTabsDidPressPrivateTab(tab: Tab?) {
         guard let selectedTab = tabManager.selectedTab else {
             return
         }
         urlBar.leaveOverlayMode()
-        if selectedTab.isPrivate {
+        if let tab = tab {
+            tabManager.selectTab(tab)
+        }
+        else if selectedTab.isPrivate {
```

Comment:
Nit: I find this `if` statement difficult to parse. It seems weird that each of these would be an exclusive case - like selecting the tab, but then checking the selected tab if it's private.